### ISSUE TITLE
Syntax Highlighting Bugs

### ISSUE THREAD

**Issue by KOGI** [benweet/stackedit#1188]
This is a really strange bug that causes incorrect highlighting in some situations. It took a while to distill this, but if you look at the middle if-block, there's a comment there about removing one of the dots on the next line. You can see how the highlighting is incorrect before removing the dot and different, but still incorrect after removing the dot. Why it affects the formatting above the dots as well as below seems strange, too.

[Link to Gist](https://stackedit.io/viewer#!provider=gist&gistId=abdc0ecf00b60547b17b033cd0f5b010&filename=Highlighting+Issue)

Screenshots
Before: ![clipboard01](https://user-images.githubusercontent.com/1231996/34580232-16e65bae-f141-11e7-815a-736c01844a24.jpg)

After: ![clipboard02](https://user-images.githubusercontent.com/1231996/34580237-1d3bb0c6-f141-11e7-98a8-ccfe3c35861d.jpg)

**EDIT**: It appears that removing the language hint (from `dart` to no hint) causes the same highlighting bug, suggesting both that `dart` isn't a supported language _and_ that the bug actually lies in the "generic" highlighting rules. Also, changing the highlighting language hint to `java` or `javascript` seems to fix the issue.

**Comment by benweet** [comment]
Please try with the v5 and create a new ticket if it's not fixed.

**Comment by KOGI** [comment]
v5 seems to have fixed it!
